crates/bevy_transform/src/hierarchy/hierarchy_maintenance_system.rs

```diff
@@ -1,30 +1,33 @@
 use crate::components::*;
-use bevy_ecs::{Commands, Entity, Query, Without};
+use bevy_ecs::{
+    core::{Entity, Without},
+    system::{Commands, Query, QuerySet},
+};
 use bevy_utils::HashMap;
 use smallvec::SmallVec;
 
 pub fn parent_update_system(
-    commands: &mut Commands,
-    removed_parent_query: Query<(Entity, &PreviousParent), Without<Parent>>,
-    // The next query could be run with a Changed<Parent> filter. However, this would mean that modifications later in the frame are lost.
-    // See issue 891: https://github.com/bevyengine/bevy/issues/891
```

Comment:
Does this still apply?

Reply:
It does. And I should actually revert most of the changes in this file. They were a reaction to using components for same-system query collision (prior to the FilteredAccess addition). FilteredAccess gives us the ability to use the old, nicer implementation.

Nice catch!

Reply:
Ahh interesting. To make this work as expected, we need to ensure reads/writes in the `Access<T>` in `FilteredAccess<T>` also sets the With flag in FilteredAccess. We should probably just expose `Access<T>` as read-only and then expose add_read/write operations on FilteredAccess